pxd.c

```diff
@@ -1615,8 +1619,8 @@ ssize_t pxd_remove(struct fuse_conn *fc, struct pxd_remove_out *remove)
 		mutex_unlock(&pxd_dev->disk->queue->sysfs_lock);
 #else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5,13,0)
-		// Do not mark queue dead, del_gendisk will try to
-		// submit all outstanding IOs on this device
+		blk_freeze_queue_start(pxd_dev->disk->queue);
+		blk_mark_disk_dead(pxd_dev->disk);
```

Comment:
del_gendisk is still getting called from this path
this can be added only after we have removed the pxd_free_disk() below

Reply:
the above are calls needed for handling surprise device removal case.
i haven't merged the async detach changes which we discussed offline.